cedar-policy/src/api.rs

```diff
@@ -1701,6 +1710,14 @@ impl PolicySet {
         }
     }
 
+    /// Remove a static `Policy` from the `PolicySet`
+    pub fn remove(&mut self, policy_id: PolicyId) -> Result<(), PolicySetError> {
+        match self.policies.remove(&policy_id) {
+            Some(_) => Ok(()),
+            None => Err(PolicySetError::RemovePolicyError(policy_id)),
+        }
```

Comment:
I think we need to also remove from `self.ast`.  (Also, do we have a test that would catch this? i.e., that confirms via an authz query that the policy was removed?)